demos/common/hfsm/az_iot_hfsm_sync_adapter.c

```diff
@@ -0,0 +1,309 @@
+ */
+
+#include "logging_levels.h"
+
+/* Logging configuration for the Demo. */
+#ifndef LIBRARY_LOG_NAME
+    #define LIBRARY_LOG_NAME    "IoT_HFSM_SYNC"
+#endif
+
+#ifndef LIBRARY_LOG_LEVEL
+    #define LIBRARY_LOG_LEVEL    LOG_INFO
+#endif
+
+/* 
+ * The function prints to the console before the network is connected;
+ * then a UDP port after the network has connected. */
+extern void vLoggingPrintf( const char * pcFormatString,
+                            ... );
+
+/* Map the SdkLog macro to the logging function to enable logging */
+#ifndef SdkLog
+    #define SdkLog( message )    vLoggingPrintf message
+#endif
+
+#include "logging_stack.h"
+/**************************************************/
+
+// The retry Hierarchical Finite State Machine object.
+// A single Provisioning + Hub client is supported when syncrhonous mode is used.
+static az_iot_hfsm_type iot_hfsm;
+
+// HFSM adapters for Provisioning and Hub operations.
+#ifdef AZ_IOT_HFSM_PROVISIONING_ENABLED
+static az_hfsm provisioning_hfsm;
+#endif
+static az_hfsm hub_hfsm;
+
+// The two HFSM objects share the same implementation. Only one is active at any time.
+#ifdef AZ_IOT_HFSM_PROVISIONING_ENABLED
+static az_hfsm* active_hfsm = &provisioning_hfsm;
+#else
+static az_hfsm* active_hfsm = &hub_hfsm;
+#endif
+
+typedef enum
+{
+  AZ_IOT_HFSM_SYNC_DO_WORK = AZ_HFSM_EVENT(1),
+  AZ_IOT_HFSM_SYNC_DO_DELAY = AZ_HFSM_EVENT(2)
+} az_iot_hfsm_sync_event_type;
+
+static const az_hfsm_event az_hfsm_event_do_work = { AZ_IOT_HFSM_SYNC_DO_WORK, NULL };
+static const az_hfsm_event az_hfsm_event_do_delay = { AZ_IOT_HFSM_SYNC_DO_DELAY, NULL };
+
+static az_hfsm_event az_hfsm_sync_event_error = { AZ_IOT_ERROR, NULL };
+
+static int32_t delay_milliseconds;
+
+static int32_t azure_iot_sync(az_hfsm* me, az_hfsm_event event);
+static int32_t idle(az_hfsm* me, az_hfsm_event event);
+static int32_t running(az_hfsm* me, az_hfsm_event event);
+static int32_t timeout(az_hfsm* me, az_hfsm_event event);
+
+static az_hfsm_state_handler provisioning_and_hub_get_parent(az_hfsm_state_handler child_state)
+{
+  az_hfsm_state_handler parent_state;
+
+  if ((child_state == azure_iot_sync))
+  {
+    parent_state = NULL;
+  }
+  else 
+  {
+    parent_state = azure_iot_sync;
+  }
+
+  return parent_state;
+}
+
+static int32_t azure_iot_sync(az_hfsm* me, az_hfsm_event event)
+{
+  switch ((int32_t)event.type)
+  {
+    case AZ_HFSM_ENTRY:
+    case AZ_HFSM_EXIT:
+      break;
+    
+    default:
+      // Control should never reach this azure_iot_sync state.
+      LogInfo( ("azure_iot_sync: PANIC!") );
+      az_iot_hfsm_pal_critical_error(me);
+  }
+
+  return 0;
+}
+
+static int32_t idle(az_hfsm* me, az_hfsm_event event)
+{
+  int32_t ret = 0;
+
+  switch ((int32_t)event.type)
+  {
+    case AZ_HFSM_ENTRY:
+    case AZ_HFSM_EXIT:
+    case AZ_IOT_HFSM_SYNC_DO_WORK:
+      break;
+
+    case AZ_IOT_START:
+        LogInfo( ("idle: AZ_IOT_START") );
```

Comment:
hfsm sync adapter idle

Reply:
The state machine name is already handled in the original logging. 
Still, I'm moving to az_log like

```C
    case AZ_IOT_START:
      if (_az_LOG_SHOULD_WRITE(AZ_LOG_IOT_START))
      {
        _az_LOG_WRITE(AZ_LOG_IOT_START, AZ_SPAN_FROM_STR("az_iot_hfsm/azure_iot/idle"));
      }
```